code/modules/mod/modules/modules_ninja.dm

```diff
@@ -0,0 +1,440 @@
+	module_type = MODULE_ACTIVE
+	use_power_cost = DEFAULT_CHARGE_DRAIN * 6
+	incompatible_modules = list(/obj/item/mod/module/energy_net)
+	cooldown_time = 1.5 SECONDS
+
+/obj/item/mod/module/energy_net/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	if(!isliving(target))
+		balloon_alert(mod.wearer, "invalid target!")
+		return
+	var/mob/living/living_target = target
+	if(locate(/obj/structure/energy_net) in get_turf(living_target))
+		balloon_alert(mod.wearer, "already trapped!")
+		return
+	for(var/turf/between_turf as anything in get_line(get_turf(mod.wearer), get_turf(living_target)))
+		if(between_turf.density)
+			balloon_alert(mod.wearer, "not through obstacles!")
+			return
+	mod.wearer.Beam(living_target, "n_beam", time = 1.5 SECONDS)
+	mod.wearer.say("Get over here!", forced = type)
+	var/obj/structure/energy_net/net = new /obj/structure/energy_net(living_target.drop_location())
+	net.affected_mob = living_target
+	mod.wearer.visible_message(span_danger("[mod.wearer] caught [living_target] with an energy net!"), span_notice("You caught [living_target] with an energy net!"))
+	if(living_target.buckled)
+		living_target.buckled.unbuckle_mob(living_target, force = TRUE)
+	net.buckle_mob(living_target, force = TRUE)
+	drain_power(use_power_cost)
+
+///Adrenaline Boost - Stops all stuns the ninja is affected with, increases his speed.
+/obj/item/mod/module/adrenaline_boost
+	name = "MOD adrenaline boost module"
+	desc = "The secrets of the Spider Clan are many. The exact specifications of their suits, \
+		the techniques they use to make every singular cut make their enemies weep with admiration, \
+		but one of their greatest mysteries is the chemical compound their assassin-saboteurs use in times of need. \
+		It's capable of clearing any fatigue whatsoever from the user, any immobilizing effect, and can even \
+		cure total paralysis. All that's known is that the fluid requires radiation to properly 'cook,' \
+		so this module demands radium to be refilled with."
+	icon_state = "adrenaline_boost"
+	removable = FALSE
+	module_type = MODULE_USABLE
+	incompatible_modules = list(/obj/item/mod/module/adrenaline_boost)
+	cooldown_time = 12 SECONDS
+	/// What reagent we need to refill?
+	var/reagent_required = /datum/reagent/uranium/radium
+	/// How much of a reagent we need to refill the boost.
+	var/reagent_required_amount = 20
+	/// Do we have a boost charge?
+	var/charged = TRUE
+
+/obj/item/mod/module/adrenaline_boost/on_use()
+	. = ..()
+	if(!.)
+		return
+	if(!charged)
+		balloon_alert(mod.wearer, "no charge!")
+		return
+	mod.wearer.SetUnconscious(0)
+	mod.wearer.SetStun(0)
+	mod.wearer.SetKnockdown(0)
+	mod.wearer.SetImmobilized(0)
+	mod.wearer.SetParalyzed(0)
+	mod.wearer.adjustStaminaLoss(-200)
+	mod.wearer.remove_status_effect(/datum/status_effect/speech/stutter)
+	mod.wearer.reagents.add_reagent(/datum/reagent/medicine/stimulants, 5)
+	mod.wearer.say(pick_list_replacements(NINJA_FILE, "lines"), forced = type)
```

Comment:
If you want to make this even more modular, it'd be great if the voice lines only trigger when the suit wearer is a ninja